src/theory/rep_set.cpp

```diff
@@ -264,7 +276,10 @@ bool RepSetIterator::initialize()
       if (d_rs->hasType(tn))
       {
         d_enum_type.push_back( ENUM_DEFAULT );
-        d_rs->getRepresentatives(tn, d_domain_elements[v]);
+        if (const auto* type_reps = d_rs->getTypeRepsOrNull(tn))
+        {
+          AppendNodesTo(*type_reps, &d_domain_elements[v]);
```

Comment:
As a matter of style, I prefer not using non-member functions and having it clear from the context what this code is doing without having to look up to find what AppendNodesTo is, thus I'd prefer if the body of AppendNodesTo was inlined here.  Is there another reason for having AppendNodesTo?

Reply:
There are a couple of reasons to have AppendNodesTo be a different function:
1. This can be standardized in a header at a later date. I have doing a header of helper functions for common stl patterns on my TODO list.
2. The name and type of the function make it fairly clear what this does IMO.
3. The order of evaluations actually becomes really iffy in the "obvious" version. `d_domain_elements[v].insert(d_domain_elements[v].end(), type_reps.begin(), type_reps.end());`. What is iffy is that `d_domain_elements[v]` has side effects and evaluation order comes into play. Coverity has a tendency to complain about iterators derived from side-effecting operators. (See cids 1362691, 1362681, 1362697, etc) To avoid this, one should probably be doing the operator[] exactly once:
```c++
std::vector<Node>& v_reps = d_domain_elements[v];
v_reps.insert(v_reps.back(), type_reps.begin(), type_reps.end());
```
This two line version is a bit less readable IMO, but it the alternative. Having an intermediary function is a nice way to force the the reference to be taken as intended. I am a bit worried someone might try to "inline" v_reps inappropriately. (It also minimizes the number of lookups.)

Concern 3 is really what I care about though. Andy, let me know if you prefer the intermediate function or the two line version.